autogluon/task/text_classification/dataset.py

```diff
@@ -233,37 +240,53 @@ def get_dataset(self, segment='train'):
         elif segment == 'dev':
             return SampledDataset(self.dataset, self.dev_sampler)
         else:
-            raise NotImplementedError
+            raise ValueError('Please specify either train or dev dataset.')
 
     def _read(self, **kwargs):
-        path = kwargs.get('filepath', None)
-        kwargs['filepath_or_buffer'] = path
-        kwargs.pop("filepath")
         low_memory = kwargs.get('low_memory', None)
         if low_memory is None:
             kwargs['low_memory'] = False
-        dataset_df = pd.read_csv(**kwargs)
+        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
+        if split_ratio is None:
+            split_ratio = 0.8
+        if 'split_ratio' in kwargs:
+            kwargs.pop('split_ratio')
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        dataset_df = kwargs.get('df', None)
+        if path is not None:
+            kwargs.pop('filepath')
+            if 'df' in kwargs:
+                kwargs.pop('df')
+            dataset_df = pd.read_csv(**kwargs)
+        elif dataset_df is not None:
+            dataset_df = dataset_df.copy(deep=True)
+        else:
+            raise ValueError('A dataset constructed from either an external csv/tsv file by setting filepath=x,'
+                             'or an in-memory python DataFrame by setting df=x is supported now.')
         dataset_df_lst = dataset_df.values.tolist()
         self.dataset = gluon.data.SimpleDataset(dataset_df_lst)
-        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
         self.train_sampler, self.dev_sampler = get_split_samplers(self.dataset, split_ratio=split_ratio)
 
     def _index_label_column(self, **kwargs):
         label_column_id = kwargs.get('label', None)
         if label_column_id is None:
+            warnings.warn('By default we are using the last column as the label column, '
+                          'if you wish to specify the label column by yourself or the column is not label column,'
+                          'please specify label column using either the column name or index by using label=x.')
             return len(self.dataset[0]) - 1
         if isinstance(label_column_id, str):
             usecols = kwargs.get('usecols', None)
             if usecols is None:
-                raise NotImplementedError
+                raise ValueError('Please specify the `usecols` when specifying the label column.')
             else:
                 for i, col in enumerate(usecols):
                     if col == label_column_id:
                         return i
         elif isinstance(label_column_id, int):
             return label_column_id
         else:
-            raise NotImplementedError
+            raise ValueError('Please specify label column using either the column name (str) or index (int).')
```

Comment:
Just a comment, does not have to be resolved in this PR: A column name can be an integer. This breaks this functionality.